src/LibraryWindow.vala

```diff
@@ -1152,19 +1158,15 @@ public class Noise.LibraryWindow : LibraryWindowInterface, Gtk.Window {
     private void on_quit () {
         if (!main_settings.privacy_mode_enabled ()) {
             // Save media position and info
-            main_settings.last_media_position = (int)((double)App.player.player.get_position
-            ()/TimeUtils.NANO_INV);
+            App.saved_state.set_int ("last-media-position", (int)((double)App.player.player.get_position()/TimeUtils.NANO_INV));
             if (App.player.current_media != null) {
                 App.player.current_media.resume_pos = (int)((double)App.player.player.get_position ()/TimeUtils.NANO_INV);
                 library_manager.update_media (App.player.current_media, false, false);
             }
-        }
-        App.player.player.pause ();
 
-        // Search
-        if (!main_settings.privacy_mode_enabled ()) {
-            main_settings.search_string = search_entry.text;
+            App.saved_state.set_string ("search-string", search_entry.text);
         }
+        App.player.player.pause ();
```

Comment:
Shouldnt this be before you read the position ?  Then, do you need to read the position twice when quitting?